File: algorithms/bubblesort/bubble_sort.go
package main

import "fmt"

func main() {
	unsorted := []int{
		9, 2, 8, 3, 1, 10, 6, 7, 4, 5,
	}
	fmt.Println(bubbleSort(unsorted))
}

func bubbleSort(array []int) []int {
	for range array {
		for j := 0; j < len(array)-1; j++ {
			if array[j] > array[j+1] {
				array[j], array[j+1] = array[j+1], array[j]
			}
		}
	}
	return array
}

File: algorithms/bubblesort/bubble_sort.py
def bubble_sort(array):
    for i in array:
        for j in range(len(array) - 1):
            if array[j] > array[j + 1]:
                array[j], array[j+1] = array[j+1], array[j]
    return array

def main():
    unsorted = [9, 2, 8, 3, 1, 10, 6, 7, 4, 5]
    print(bubble_sort(unsorted))

if __name__ == "__main__":
    main()
